src/main/java/io/cryostat/agent/remote/RecordingsContext.java

```diff
@@ -71,67 +103,147 @@ public String path() {
+        boolean restricted = blocked.contains(mtd);
+        if (!restricted) {
+            return true;
         }
+        boolean passed = restricted && MutatingRemoteContext.apiWritesEnabled(config);
+        if (!passed) {
+            exchange.sendResponseHeaders(HttpStatus.SC_FORBIDDEN, -1);
+        }
+        return passed;
     }
 
-    private List<RecordingInfo> getRecordings() {
+    private List<SerializableRecordingDescriptor> getRecordings() {
         return FlightRecorder.getFlightRecorder().getRecordings().stream()
-                .map(RecordingInfo::new)
+                .map(SerializableRecordingDescriptor::new)
                 .collect(Collectors.toList());
     }
 
-    @SuppressFBWarnings(value = "URF_UNREAD_FIELD")
-    private static class RecordingInfo {
-
-        public final long id;
-        public final String name;
-        public final String state;
-        public final Map<String, String> options;
-        public final long startTime;
-        public final long duration;
-        public final boolean isContinuous;
-        public final boolean toDisk;
-        public final long maxSize;
-        public final long maxAge;
-
-        RecordingInfo(Recording rec) {
-            this.id = rec.getId();
-            this.name = rec.getName();
-            this.state = rec.getState().name();
-            this.options = rec.getSettings();
-            if (rec.getStartTime() != null) {
-                this.startTime = rec.getStartTime().toEpochMilli();
-            } else {
-                this.startTime = 0;
-            }
-            this.isContinuous = rec.getDuration() == null;
-            this.duration = this.isContinuous ? 0 : rec.getDuration().toMillis();
-            this.toDisk = rec.isToDisk();
-            this.maxSize = rec.getMaxSize();
-            if (rec.getMaxAge() != null) {
-                this.maxAge = rec.getMaxAge().toMillis();
+    private SerializableRecordingDescriptor startRecording(StartRecordingRequest req)
+            throws QuantityConversionException, ServiceNotAvailableException,
+                    FlightRecorderException,
+                    org.openjdk.jmc.rjmx.services.jfr.FlightRecorderException,
+                    InvalidEventTemplateException, InvalidXmlException, IOException {
+        Runnable cleanup = () -> {};
+        try {
+            JFRConnection conn =
+                    jfrConnectionToolkit.connect(
+                            jfrConnectionToolkit.createServiceURL("localhost", 0));
+            IConstrainedMap<EventOptionID> events;
+            if (req.requestsCustomTemplate()) {
+                Template template =
+                        localStorageTemplateService.addTemplate(
+                                new ByteArrayInputStream(
+                                        req.template.getBytes(StandardCharsets.UTF_8)));
+                events = localStorageTemplateService.getEvents(template).orElseThrow();
```

Comment:
Should this throw an `InvalidEventTemplateException`? Along with the code on line 202.